activesupport/lib/active_support/duration.rb

```diff
@@ -180,13 +180,15 @@ def since(time = ::Time.current)
       sum(1, time)
     end
     alias :from_now :since
+    alias :after :since
 
     # Calculates a new Time or Date that is as far in the past
     # as this Duration represents.
     def ago(time = ::Time.current)
       sum(-1, time)
     end
     alias :until :ago
+    alias :before :ago
```

Comment:
The process is to allow both old and new method signatures and give a deprecation warning if we see the old one.

```ruby
# Calculates this duration in the past relative to +date_or_time+.
def before(date_or_time)
  sum(-1, date_or_time)
end
alias :until :before

def ago(deprecated_date_or_time = nil)
  if deprecated_date_or_time
    ActiveSupport::Deprecation.warn "Passing a date_or_time argument to #ago is deprecated and will be removed in Rails 5.2. Use #before(date_or_time) instead."
  end
  before deprecated_date_or_time || ::Time.current
end
```